apps/apps.c

```diff
@@ -2654,65 +2654,38 @@ static const char *modeverb(char mode)
 }
 
 /*
- * Open a file for writing, owner-read-only.
+ * Open a file for writing, owner-read-only. The file is written in
+ * a 'write-to-temp-file-then-rename' basis.
  */
 BIO *bio_open_owner(const char *filename, int format, int private)
 {
-    FILE *fp = NULL;
     BIO *b = NULL;
-    int fd = -1, bflags, mode, textmode;
+    BIO *sink = NULL;
+    BIO *filter = NULL;
 
     if (!private || filename == NULL || strcmp(filename, "-") == 0)
         return bio_open_default(filename, 'w', format);
 
-    mode = O_WRONLY;
-#ifdef O_CREAT
-    mode |= O_CREAT;
-#endif
-#ifdef O_TRUNC
-    mode |= O_TRUNC;
-#endif
-    textmode = istext(format);
-    if (!textmode) {
-#ifdef O_BINARY
-        mode |= O_BINARY;
-#elif defined(_O_BINARY)
-        mode |= _O_BINARY;
-#endif
-    }
+    sink = BIO_new(BIO_s_file());
+    if (sink == NULL)
+        goto err;
 
-#ifdef OPENSSL_SYS_VMS
-    /* VMS doesn't have O_BINARY, it just doesn't make sense.  But,
-     * it still needs to know that we're going binary, or fdopen()
-     * will fail with "invalid argument"...  so we tell VMS what the
-     * context is.
-     */
-    if (!textmode)
-        fd = open(filename, mode, 0600, "ctx=bin");
-    else
-#endif
-        fd = open(filename, mode, 0600);
-    if (fd < 0)
+    /* create the filter BIO */
+    filter = BIO_new(apps_bf_tempfile());
+    if (filter == NULL)
         goto err;
-    fp = fdopen(fd, modestr('w', format));
-    if (fp == NULL)
+
+    b = BIO_push(filter, sink);
+
+    /* set real filename to filter BIO for later usage */
+    if (!BIO_ctrl(filter, BIO_C_SET_FILENAME, format, (void *)filename))
```

Comment:
You could use `b` instead of `filter` ;-)
Even though they are the same, I think `b` is more intuitive to read.

Reply:
Yeah...